CMakeLists.txt

```diff
@@ -1,26 +1,82 @@
-cmake_minimum_required (VERSION 3.14)
-project (ifc-sdk)
+# Copyright Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-# Fetch Microsoft GSL
-include(FetchContent)
+cmake_minimum_required(VERSION 3.14)
 
-FetchContent_Declare(GSL
-    GIT_REPOSITORY "https://github.com/microsoft/GSL"
-    GIT_TAG "v4.0.0"
+project(
+    ifc-sdk
+    VERSION 0.0.1
+    LANGUAGES CXX
 )
-FetchContent_MakeAvailable(GSL)
 
-if (MSVC)
-add_compile_options(
-    /EHsc          # Turn on exception handling semantics
+option(DEVELOPER_MODE "Enable project options and targets for developers of ifc-sdk" OFF)
+mark_as_advanced(DEVELOPER_MODE)
+include(CMakeDependentOption)
+
+find_package(Microsoft.GSL REQUIRED)
+
```

Comment:
CMake will fail loudly if GSL is not found. The user has to provide it. vcpkg.json is provided for convenience, but package managers (the people) will usually know better.

Reply:
Yeah, that is what I thought.  It would be useful if the configuration would fetch GSL if wasn't installed (the current behavior).

Reply:
The current behavior is broken, as it isn't depending of the upstream install interface. The GSL package exports the target `Microsoft.GSL::GSL`, but that is nowhere to be found in the code. This would make it very difficult to package this project.

One thing that could be done without breaking best practices is to make use of the fact that `find_package` (the ultimate way of importing dependencies) is to make use of the fact that it's practically just an `include` with a lot of conventional features. [This example](https://github.com/friendlyanon/cmake-init-fetchcontent) illustrates how opting into FetchContent can be done with realitve ease.

By default, I would definitely prefer nudging people in the direction of package managers though. vcpkg is "in house" as far as this project is concerned so I view that as a bonus :)

Reply:
> vcpkg is "in house" 

It is true that vcpkg is in house but we are not requiring it for this project.  We do however want to take advantage of it when it is available.

> [This example](https://github.com/friendlyanon/cmake-init-fetchcontent) illustrates how opting into FetchContent can be done with realitve ease.

A solution with "opting" into FetchContent with relative ease is good to explore.